lib/mp4/probe.js

```diff
@@ -221,10 +221,106 @@ getVideoTrackIds = function(init) {
   return videoTrackIds;
 };
 
+getTracks = function(init) {
+  var traks = findBox(init, ['moov', 'trak']);
+  var tracks = [];
+
+  traks.forEach(function(trak) {
+    var track = {};
+    var tkhd = findBox(trak, ['tkhd'])[0];
+
+    // id
+    if (tkhd) {
+      var view = new DataView(tkhd.buffer, tkhd.byteOffset, tkhd.byteLength);
+      var version = view.getUint8(0);
+
+      track.id = (version === 0) ? view.getUint32(12) : view.getUint32(20);
+    }
+
+    var hdlr = findBox(trak, ['mdia', 'hdlr'])[0];
+
+    // type
+    if (hdlr) {
+      var type = parseType(hdlr.subarray(8, 12));
+
+      if (type === 'vide') {
+        track.type = 'video';
+      } else if (type === 'soun') {
+        track.type = 'audio';
+      } else {
+        track.type = type;
+      }
+    }
+
+
+    // codec
+    var stsd = findBox(trak, ['mdia', 'minf', 'stbl', 'stsd'])[0];
+
+    if (stsd) {
+      var sampleDescriptions = stsd.subarray(8);
+      // gives the codec type string
+      track.codec = parseType(sampleDescriptions.subarray(4, 8));
+
+      // https://tools.ietf.org/html/rfc6381#section-3.3
+      if ((/^[a-z]vc[1-9]$/i).test(track.codec)) {
+        var codecConfig = sampleDescriptions.subarray(94);
+        // always avcc
+        // var type = parseType(codecConfig.subarray(4, 8));
+
+        if (codecConfig.length > 11) {
+          track.codec += '.';
+
+          // left padded with zeroes for single digit hex
+          // profile idc
+          track.codec +=  ('00' + codecConfig[9].toString(16)).slice(-2);
+          // the byte containing the constraint_set flags
+          track.codec += ('00' + codecConfig[10].toString(16)).slice(-2);
+          // level idc
+          track.codec += ('00' + codecConfig[11].toString(16)).slice(-2);
+        }
+      } else if ((/^mp4[a,v]$/i).test(track.codec)) {
+        var codecConfig = sampleDescriptions.subarray(36);
+        // always esds
+        // var type = parseType(codecConfig.subarray(4, 8));
+
+        if (codecConfig.length > 20) {
+          track.codec += '.' + ('00' + codecConfig[19].toString(16)).slice(-2);
+          track.codec += '.' + (((codecConfig[20] >>> 2) & 0x3f).toString(16)).slice(-2);
+        }
+      } else {
+        // TODO: show a warning?
+      }
+    }
+
+    // TODO: should we parse timescale here?
```

Comment:
I think we parse the timescales again later on the in the process using https://github.com/videojs/mux.js/blob/d8a5cd7a2a5fbe82b574dd47ea074d8f957987db/lib/mp4/probe.js#L86

Reply:
So the referenced function gets the timescale for all tracks. This function gets all the relevant information about every track including its timescale and stores it in a `track` object. I think that it makes more sense to do it that way. The TODO is mainly a question of wether we should track the `startTime` of each track separately or not, using their own timescale.

Reply:
hmm, considering we're bothering to get the timescales in this function, might as well also get the `startTime`s for each track as well, so we can get all the information back in one call

Reply:
we can only get the start time from an mp4 fragment, but we get the timescales from the mp4 init segment, so we have to get them at separate.

Reply:
seems like you answered your TODO then 😄

Reply:
We cannot, as we get `startTime` from mp4 fragments, but we get timescale from the regular mp4 init segment.